app.go

```diff
@@ -787,11 +788,19 @@ func (app *App) Stop(ctx context.Context) (err error) {
 // using the Shutdown functionality (see the Shutdowner documentation for details).
 func (app *App) Done() <-chan os.Signal {
 	c := make(chan os.Signal, 1)
-	signal.Notify(c, _sigINT, _sigTERM)
 
 	app.donesMu.Lock()
+	defer app.donesMu.Unlock()
+	// If shutdown signal has been received already
+	// send it and return. If not, wait for user to send a termination
+	// signal.
+	if app.shutdownSig != nil {
+		c <- app.shutdownSig
+		return c
+	}
+	signal.Notify(c, _sigINT, _sigTERM)
+
```

Comment:
nit: signal.Notify is part of the second "paragraph" in that the two blocks are "already shutdown" and "not yet shutdown" so the statements for those should be grouped together.
```suggestion

	signal.Notify(c, _sigINT, _sigTERM)
```